packages/donor/src/components/AppointmentSelect/AppointmentSelect.tsx

```diff
@@ -0,0 +1,171 @@
+import { Player } from "@lottiefiles/react-lottie-player";
+import {
+  AvailableAppointment,
+  BookedAppointment,
+  Hospital,
+  HospitalUtils,
+} from "@zm-blood-components/common";
+import { useState } from "react";
+import { DonationSlotToBook } from "../../state/AppointmentToBookStore";
+import {
+  DonationDay,
+  groupDonationDays,
+} from "../../utils/AppointmentsGrouper";
+import AppointmentPicker from "../AppointmentPicker";
+import Popup, { PopupProps } from "../basic/Popup";
+import Select from "../basic/Select";
+import styles from "./AppointmentSelect.module.scss";
+import WarningLogo from "../../assets/images/warning.svg";
+import React from "react";
+import DropAnimation from "../../assets/animations/drop.json";
+import NoAppointments from "../../assets/images/NO Appointments.svg";
+
+export interface AppointmentSelectProps {
+  availableAppointments: AvailableAppointment[];
+  isFetching: boolean;
+  onSlotSelected: (donationSlot: DonationSlotToBook) => void;
+  tooCloseDonationPopupProps: Pick<
+    PopupProps,
+    "open" | "onBack" | "onApproved" | "onClose"
+  >;
+  appointmentToHide?: BookedAppointment;
+  isSwapAppointment: boolean;
+}
+
+export default function AppointmentSelect({
+  availableAppointments,
+  isFetching,
+  onSlotSelected,
+  tooCloseDonationPopupProps,
+  appointmentToHide,
+  isSwapAppointment,
+}: AppointmentSelectProps) {
+  const [selectedHospital, setSelectedHospital] = useState<Hospital | "">("");
+
+  const sortedDonationDays = React.useMemo(() => {
+    const appointmentFilter = (appointment: AvailableAppointment) => {
+      const isValidHospital =
+        appointment.hospital === selectedHospital || !selectedHospital;
+
```

Comment:
if isValidHospital == false, there's no point to continue, just return false.

Reply:
Done